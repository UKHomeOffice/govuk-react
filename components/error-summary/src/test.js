import React from 'react';
import ReactDOM from 'react-dom';
import ErrorSummary from './';

describe(ErrorSummary, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ErrorSummary>Example</ErrorSummary>, div);
  });
});
