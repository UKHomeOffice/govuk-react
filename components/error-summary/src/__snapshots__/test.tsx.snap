// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error summary matches the ErrorSummary snapshot: error summary 1`] = `
<ErrorSummaryExample>
  <div>
    <ErrorSummary
      description="Optional description of the errors and how to correct them"
      errors={
        Array [
          Object {
            "targetName": "national-insurance-number",
            "text": "National Insurance number error",
          },
          Object {
            "targetName": "description",
            "text": "Description of what you saw error",
          },
        ]
      }
      heading="Message to alert the user to a problem goes here"
      onHandleErrorClick={[Function]}
    >
      <src__StyledErrorSummary
        tabIndex={-1}
      >
        <div
          className="src__StyledErrorSummary-sc-6d0w2-1 hFfdPr"
          tabIndex={-1}
        >
          <H2
            size="MEDIUM"
          >
            <Heading
              as="h2"
              size="MEDIUM"
            >
              <heading__StyledHeading
                as="h2"
                size="MEDIUM"
              >
                <h2
                  className="heading__StyledHeading-sc-1dsobun-0 cSTIAh"
                  size="MEDIUM"
                >
                  Message to alert the user to a problem goes here
                </h2>
              </heading__StyledHeading>
            </Heading>
          </H2>
          <Paragraph
            linkRenderer={[Function]}
            mb={3}
            supportingText={false}
          >
            <src__StyledParagraph
              allowedTypes={
                Array [
                  "paragraph",
                  "emphasis",
                  "strong",
                  "link",
                  "inlineCode",
                  "code",
                  "text",
                ]
              }
              escapeHtml={false}
              linkRenderer={[Function]}
              mb={3}
              renderers={
                Object {
                  "link": [Function],
                }
              }
              skipHtml={true}
              source="Optional description of the errors and how to correct them"
              supportingText={false}
            >
              <ReactMarkdown
                allowedTypes={
                  Array [
                    "paragraph",
                    "emphasis",
                    "strong",
                    "link",
                    "inlineCode",
                    "code",
                    "text",
                  ]
                }
                className="src__StyledParagraph-sgc9po-0 ilsyuQ"
                escapeHtml={false}
                linkRenderer={[Function]}
                mb={3}
                renderers={
                  Object {
                    "link": [Function],
                  }
                }
                skipHtml={true}
                source="Optional description of the errors and how to correct them"
                supportingText={false}
                transformLinkUri={[Function]}
              >
                <Root
                  className="src__StyledParagraph-sgc9po-0 ilsyuQ"
                  key="root-1-1-0"
                  node={
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "position": Object {
                                "end": Object {
                                  "column": 59,
                                  "line": 1,
                                  "offset": 58,
                                },
                                "start": Object {
                                  "column": 1,
                                  "line": 1,
                                  "offset": 0,
                                },
                              },
                              "type": "text",
                              "value": "Optional description of the errors and how to correct them",
                            },
                          ],
                          "position": Object {
                            "end": Object {
                              "column": 59,
                              "line": 1,
                              "offset": 58,
                            },
                            "start": Object {
                              "column": 1,
                              "line": 1,
                              "offset": 0,
                            },
                          },
                          "type": "paragraph",
                        },
                      ],
                      "position": Object {
                        "end": Object {
                          "column": 59,
                          "line": 1,
                          "offset": 58,
                        },
                        "start": Object {
                          "column": 1,
                          "line": 1,
                          "offset": 0,
                        },
                      },
                      "type": "root",
                    }
                  }
                >
                  <div
                    className="src__StyledParagraph-sgc9po-0 ilsyuQ"
                  >
                    <p
                      key="paragraph-1-1-0"
                    >
                      <TextRenderer
                        key="text-1-1-0"
                        node={
                          Object {
                            "position": Object {
                              "end": Object {
                                "column": 59,
                                "line": 1,
                                "offset": 58,
                              },
                              "start": Object {
                                "column": 1,
                                "line": 1,
                                "offset": 0,
                              },
                            },
                            "type": "text",
                            "value": "Optional description of the errors and how to correct them",
                          }
                        }
                        nodeKey="text-1-1-0"
                        value="Optional description of the errors and how to correct them"
                      >
                        Optional description of the errors and how to correct them
                      </TextRenderer>
                    </p>
                  </div>
                </Root>
              </ReactMarkdown>
            </src__StyledParagraph>
          </Paragraph>
          <UnorderedList
            listStyleType="none"
            mb={0}
          >
            <src__OrderedList
              as="ul"
              listStyleType="none"
              mb={0}
            >
              <ul
                className="src__OrderedList-sc-1392v8j-0 kvLUna"
              >
                <src__ListItem
                  key="national-insurance-number"
                >
                  <li
                    className="src__ListItem-u4zf5o-0 caTzFj"
                  >
                    <src__StyledErrorText
                      muted={false}
                      noVisitedState={false}
                      onClick={[Function]}
                      tabIndex={1}
                      textColour={false}
                    >
                      <a
                        className="src__StyledErrorText-sc-6d0w2-0 hKXFDy hgczep"
                        onClick={[Function]}
                        tabIndex={1}
                      >
                        National Insurance number error
                      </a>
                    </src__StyledErrorText>
                  </li>
                </src__ListItem>
                <src__ListItem
                  key="description"
                >
                  <li
                    className="src__ListItem-u4zf5o-0 caTzFj"
                  >
                    <src__StyledErrorText
                      muted={false}
                      noVisitedState={false}
                      onClick={[Function]}
                      tabIndex={2}
                      textColour={false}
                    >
                      <a
                        className="src__StyledErrorText-sc-6d0w2-0 hKXFDy hgczep"
                        onClick={[Function]}
                        tabIndex={2}
                      >
                        Description of what you saw error
                      </a>
                    </src__StyledErrorText>
                  </li>
                </src__ListItem>
              </ul>
            </src__OrderedList>
          </UnorderedList>
        </div>
      </src__StyledErrorSummary>
    </ErrorSummary>
    <InputField
      hint="It’s on your National Insurance card, benefit letter, payslip or P60. For example, ‘QQ 12 34 56 C’."
      input={Object {}}
      meta={Object {}}
      name="national-insurance-number"
    >
      <Label
        error={false}
        name="national-insurance-number"
      >
        <src__StyledLabel
          error={false}
          name="national-insurance-number"
        >
          <label
            className="src__StyledLabel-iqzvxn-0 jquVEk"
            name="national-insurance-number"
          >
            <LabelText>
              <src__StyledLabelText>
                <span
                  className="src__StyledLabelText-sc-1lbxenh-0 hiKxXi"
                >
                  National Insurance number
                </span>
              </src__StyledLabelText>
            </LabelText>
            <HintText>
              <src__StyledHint>
                <span
                  className="src__StyledHint-tq1z5r-0 bBwCby"
                >
                  It’s on your National Insurance card, benefit letter, payslip or P60. For example, ‘QQ 12 34 56 C’.
                </span>
              </src__StyledHint>
            </HintText>
            <Input
              error={false}
              type="text"
            >
              <src__StyledInput
                error={false}
                type="text"
              >
                <input
                  className="src__StyledInput-sc-1ch9crp-0 hCAyZH"
                  type="text"
                />
              </src__StyledInput>
            </Input>
          </label>
        </src__StyledLabel>
      </Label>
    </InputField>
    <br />
    <TextArea
      input={Object {}}
      meta={Object {}}
      name="description"
    >
      <Label
        error={false}
        name="description"
      >
        <src__StyledLabel
          error={false}
          name="description"
        >
          <label
            className="src__StyledLabel-iqzvxn-0 jquVEk"
            name="description"
          >
            <LabelText>
              <src__StyledLabelText>
                <span
                  className="src__StyledLabelText-sc-1lbxenh-0 hiKxXi"
                >
                  Description of what you saw
                </span>
              </src__StyledLabelText>
            </LabelText>
            <src__TextAreaField
              rows="5"
              type="text"
            >
              <textarea
                className="src__TextAreaField-z0ua7p-0 dRWihc"
                rows="5"
                type="text"
              />
            </src__TextAreaField>
          </label>
        </src__StyledLabel>
      </Label>
    </TextArea>
  </div>
</ErrorSummaryExample>
`;
